harper-tree-sitter/src/lib.rs

```diff
@@ -91,7 +91,7 @@ impl TreeSitterMasker {
 }
 
 impl Masker for TreeSitterMasker {
-    fn create_mask(&mut self, source: &[char]) -> Mask {
+    fn create_mask(&self, source: &[char]) -> Mask {
```

Comment:
Why did you remove mutability? I remember discussing it with you, but forgot why.